File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewwStationNode : public rclcpp::Node
{
public:
  RobotNewwStationNode():Node("cpp_test"), robot_name_("AA30")
  {
    publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);

    RCLCPP_INFO(this->get_logger(), "HellO, this is AA30");
    timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                      std::bind(&RobotNewwStationNode::timerCallback, this));
  }
  
private:
  void timerCallback()
  {
    this->publishNews();
  }

  rclcpp::TimerBase::SharedPtr timer_;

  std::string robot_name_;
  rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;

  void publishNews()
  {
    auto msg = example_interfaces::msg::String();
    msg.data = "Hi, this is " + this->robot_name_;
    this->publisher_->publish(msg);
  }
};


int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<RobotNewwStationNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewStationNode(Node):
  def __init__(self):
    super().__init__("robot_news_station")
    self.declare_parameter("robot_name", "C3P0")
    
    self.robot_name_ = self.get_parameter("robot_name").value
    self.publisher_ = self.create_publisher(String, "robot_news", 10)
    self.timer_ = self.create_timer(1, self.timerCallback)
    self.get_logger().info("Robot News Station has been started")
    
  def publish_news(self):
    msg = String()
    msg.data = "Hi, this is " + self.robot_name_ + " from the robit news station."
    self.publisher_.publish(msg)
    
  def timerCallback(self):
    self.publish_news()
  
def main(args=None):
  rclpy.init(args=args)
  node = RobotNewStationNode()
  rclpy.spin(node)
  rclpy.shutdown()
  
if __name__ == "__main__":
  main()